Sources/Filters/Sources/CubeSource/index.js

```diff
@@ -191,47 +218,32 @@ function vtkCubeSource(publicAPI, model) {
       .translate(...model.center)
       .apply(points);
 
-    // Define quads
-    const polys = new Uint16Array(numberOfPolys * 5);
-    polyData.getPolys().setData(polys, 1);
-
-    let polyIndex = 0;
-
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 0;
-    polys[polyIndex++] = 1;
-    polys[polyIndex++] = 3;
-    polys[polyIndex++] = 2;
-
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 6;
-    polys[polyIndex++] = 7;
-    polys[polyIndex++] = 5;
-
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 8;
-    polys[polyIndex++] = 10;
-    polys[polyIndex++] = 11;
-    polys[polyIndex++] = 9;
-
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 12;
-    polys[polyIndex++] = 13;
-    polys[polyIndex++] = 15;
-    polys[polyIndex++] = 14;
-
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 16;
-    polys[polyIndex++] = 18;
-    polys[polyIndex++] = 19;
-    polys[polyIndex++] = 17;
-
-    polys[polyIndex++] = 4;
-    polys[polyIndex++] = 20;
-    polys[polyIndex++] = 21;
-    polys[polyIndex++] = 23;
-    polys[polyIndex] = 22;
+    // Apply optional additionally specified matrix transformation
+    vtkMatrixBuilder.buildFromRadian().setMatrix(model.matrix).apply(points);
```

Comment:
For performance reason, do not call `setMatrix` if `model.matrix` is `undefined` or `null` (should be the default).
(and maybe it should also be the same if `model.rotations` are zeros...

Reply:
Thanks. Please see the suggested changes here: https://github.com/Kitware/vtk-js/pull/2895